blaze/server/server.py

```diff
@@ -70,6 +85,12 @@ class Server(object):
         A dictionary mapping dataset name to any data format that blaze
         understands.
 
+    formats : ``iterable[SerializationFormat]``, optional
+        An iterable of supported serialization formats. By default, the
+        server will support JSON.
+        A serialization format is an object that supports:
+        name, loads, and dumps.
```

Comment:
If I were doing the work I would probably skip making the new type and use a dict

```python
formats = {'json': (json.dumps, json.loads),
           'pickle': (partial(pickle.dumps, protocol=pickle.HIGHEST_PROTOCOL), pickle.loads),
           'msgpack': (msgpack.packb, msgpack.unpackb)
           }
```

But I'm quite grateful at not having to do this work and the chosen strategy currently in this PR seems sane to me; so I say go for it, particularly since this PR has been going on for a while.